odbc/src/api/data.rs

```diff
@@ -1340,12 +1340,11 @@ mod unit {
                             $input
                         );
                         if $info.is_some() {
-                            let odbc_info = r.1.unwrap().get_sql_state().odbc_3_state.to_string();
```

Comment:
Initially, I attempted to only update this line to:
```
let odbc_info = r.1.unwrap().get_sql_state();
```

However, that led to the error:
```
error[E0716]: temporary value dropped while borrowed
    --> odbc/src/api/data.rs:1343:46
     |
1343 |                             let odbc_state = r.1.unwrap().get_sql_state();
     |                                              ^^^^^^^^^^^^                - temporary value is freed at the end of this statement
     |                                              |
     |                                              creates a temporary value which is freed while still in use
...
1539 |                 test_it!(bson, v);
     |                 -----------------
     |                 |
     |                 borrow later used here
     |                 in this macro invocation
     |
     = note: consider using a `let` binding to create a longer lived value
```
Which for the life of me I could not figure out how to get around since we're 2 macros and 2 iterators deep at this point. I decided to go the easy route by dropping the local variable completely and introducing cloning. Since this is just a set of unit tests, this seems ok to do. I am open to addressing this a different way if someone sees how to do so without cloning `r.1`.

Reply:
I should update the coding guidelines to say that unnecessary clones in tests are fine :joy:

Reply:
This makes total sense, I think this is exactly the crossroads I had come to and wanted to defer experimenting more at the time. Looks good getting rid of all of the suffixes 😄